File: app/src/main/java/com/example/calculator/StorageThemes.java
package com.example.calculator;

import android.content.Context;
import android.content.SharedPreferences;

public class StorageThemes {

    private static final String KEY_APP_THEMES = "KEY_APP_THEMES";

    private SharedPreferences pref;

    public StorageThemes(Context context) {
        pref = context.getSharedPreferences("app_themes", Context.MODE_PRIVATE);
    }

    public ApplicationThemes getThemes() {
        String key = pref.getString(KEY_APP_THEMES, ApplicationThemes.ONE_THEMES.getKey());

        for (ApplicationThemes themes :
                ApplicationThemes.values()) {
            if (themes.getKey().equals(key)) {
                return themes;
            }
        }
        throw new IllegalStateException("STUB!");
    }

    public void setThemes(ApplicationThemes themes) {
        pref.edit().putString(KEY_APP_THEMES, themes.getKey()).apply();
    }
}

File: app/src/main/java/com/example/calculator/StorageThemes.kt
package com.example.calculator

import android.content.Context
import android.content.SharedPreferences

class StorageThemes(context: Context) {

    private val pref: SharedPreferences =
        context.getSharedPreferences("app_themes", Context.MODE_PRIVATE)

    var themes: ApplicationThemes
        get() {
            val key = pref.getString(KEY_APP_THEMES, ApplicationThemes.ONE_THEMES.key)
            for (themes in ApplicationThemes.values()) {
                if (themes.key == key) {
                    return themes
                }
            }
            throw IllegalStateException("STUB!")
        }
        set(themes) {
            pref.edit().putString(KEY_APP_THEMES, themes.key).apply()
        }

    companion object {
        private const val KEY_APP_THEMES = "KEY_APP_THEMES"
    }
}